### PR TITLE
Add arm64 architecture support for buf

### PR DESCRIPTION
**Pull request by tandel-pratik**
Issue described in #192

### CODE REVIEW

File: repositories.bzl [hunk 1]
```diff
@@ -73,27 +74,39 @@ VERSIONS = {
...
-        "urls": ["https://github.com/bufbuild/buf/releases/download/v0.56.0/protoc-gen-buf-lint-Darwin-x86_64"],
+        "urls": ["https://github.com/bufbuild/buf/releases/download/{}/protoc-gen-buf-lint-Darwin-x86_64".format(BUF_VERSION)],
         "sha256": "3ff939636e5857f6fe3dcaeae816538fcee41cec66b10b62df5ccb65d0f79e7f",
         "executable": True,
     },
     "protoc_gen_buf_lint_linux_x86_64": {
         "type": "http_file",
-        "urls": ["https://github.com/bufbuild/buf/releases/download/v0.56.0/protoc-gen-buf-lint-Linux-x86_64"],
+        "urls": ["https://github.com/bufbuild/buf/releases/download/{}/protoc-gen-buf-lint-Linux-x86_64".format(BUF_VERSION)],
         "sha256": "a7ab67a5bcc5906366bde424ba63fdcf604e07d4825e5720c8e5b3ab1530bbf7",
```
Comment (tandel-pratik) [inline repositories.bzl:110]: the sha's need to be updated.

File: repositories.bzl [hunk 2]
```diff
@@ -73,27 +74,39 @@ VERSIONS = {
     },
 
     # Buf
+    "protoc_gen_buf_breaking_darwin_arm64": {
+        "type": "http_file",
+        "urls": ["https://github.com/bufbuild/buf/releases/download/{}/protoc-gen-buf-breaking-Darwin-x86_64".format(BUF_VERSION)],
```
Comment (jrusso1020) [inline repositories.bzl:79]:
```suggestion
        "urls": ["https://github.com/bufbuild/buf/releases/download/{}/protoc-gen-buf-breaking-Darwin-arm64".format(BUF_VERSION)],
```

File: repositories.bzl [hunk 3]
```diff
@@ -73,27 +74,39 @@ VERSIONS = {
...
+        "sha256": "d7b12a2ccd663f00a068b19cbd2c1e81f4983ea33bd9a92980485e2c4693b75a",
+        "executable": True,
+    },
     "protoc_gen_buf_breaking_darwin_x86_64": {
         "type": "http_file",
-        "urls": ["https://github.com/bufbuild/buf/releases/download/v0.56.0/protoc-gen-buf-breaking-Darwin-x86_64"],
+        "urls": ["https://github.com/bufbuild/buf/releases/download/{}/protoc-gen-buf-breaking-Darwin-arm64".format(BUF_VERSION)],
```
Comment (jrusso1020) [inline repositories.bzl:85]:
```suggestion
        "urls": ["https://github.com/bufbuild/buf/releases/download/{}/protoc-gen-buf-breaking-Darwin-x86_64".format(BUF_VERSION)],
```

**Review by jrusso1020 — COMMENTED** [review]
I think a couple of the urls had conflicting arch's

**Pull request merged**